jax/_src/nn/initializers.py

```diff
@@ -48,7 +48,67 @@ def _compute_fans(shape: core.NamedShape, in_axis=-2, out_axis=-1):
   fan_out = shape[out_axis] * receptive_field_size
   return fan_in, fan_out
 
+def _get_real_dtype(dtype):
+  if dtype == jnp.complex64:
+    return jnp.float32
+  elif dtype == jnp.complex128:
+    return jnp.float64
+  else:
+    return dtype
+
+def _complex_uniform(key, shape, dtype):
```

Comment:
It's possible that one or both of these should eventually live in jax.random, but here is a reasonable place for them for now.

Reply:
There was some discussions #4805, and we decided not to add them to `jax.random` for now.

Reply:
I agree that consensus lacks for `complex_uniform`, though I think that `truncated_normal` could be moved to `jax.random` as there is consensus on what that means for complex numbers.